### ISSUE TITLE
getTree from root doesn't get nested resources

### ISSUE THREAD

**Issue by modxbot** [modxcms/revolution#8430]
ben_omycode created Redmine issue ID 8430

Hi,

I tried to get the complete resource tree from root using :

<pre>
$modx->getTree(0, 10);
</pre>

but I only get the level 1 resources.

However, getTree successfully returns the tree of a specific resource which id different of 0.

The problem is the same with : 

<pre>
$modx->getChildIds(0, 10);
</pre>

**Comment by ArekX** [comment]
Wow. I cannot believe that this error still exists in version v2.5.0-rc2 even. Guys, almost 4 years passed and nobody got around to fixing such an important core class bug? 

Here's the problem inside modx.class.php:
```php
    public function getTree($id= null, $depth= 10, array $options = array()) {
        $tree= array ();
        $context = '';
        if (!empty($options['context'])) {
            $this->getContext($options['context']);
            $context = $options['context'];
        }
        if ($id !== null) {
            if (is_array ($id)) {
                foreach ($id as $k => $v) {
                    $tree[$v]= $this->getTree($v, $depth, $options);
                }
            }
            elseif ($branch= $this->getChildIds($id, 1, $options)) {
                foreach ($branch as $key => $child) {
                    if ($depth > 0 && $leaf= $this->getTree($child, $depth--, $options)) {
                        $tree[$child]= $leaf;
                    } else {
                        $tree[$child]= $child;
                    }
                }
            }
        }
        return $tree;
    }
```

`$depth` is handled incorrectly, first problem is here:
```php
 if ($depth > 0 && $leaf= $this->getTree($child, $depth--, $options)) {
```
`$depth--` is a **POST** decrement, so all first getTree will get the same depth value as the current one and the one inside it, and so on recursively...

Second problem appears after changing this to `--$depth`:
```php
 $tree[$v]= $this->getTree($v, $depth, $options);
```
As this grabs already decremented depth by other foreach it will mostly never get its results.

The fix is to simply change all both of these to `$depth - 1`

i.e. This:
```php
    public function getTree($id= null, $depth= 10, array $options = array()) {
        $tree= array ();
        $context = '';
        if (!empty($options['context'])) {
            $this->getContext($options['context']);
            $context = $options['context'];
        }
        if ($id !== null) {
            if (is_array ($id)) {
                foreach ($id as $k => $v) {
                    $tree[$v]= $this->getTree($v, $depth - 1, $options);
                }
            }
            elseif ($branch= $this->getChildIds($id, 1, $options)) {
                foreach ($branch as $key => $child) {
                    if ($depth > 0 && $leaf= $this->getTree($child, $depth - 1, $options)) {
                        $tree[$child]= $leaf;
                    } else {
                        $tree[$child]= $child;
                    }
                }
            }
        }
        return $tree;
    }
```

No need for credits. Just fix this please. :)

**Comment by Mark-H** [comment]
Would you mind sending a pull request? Instructions on how to contribute are in the contribute.MD file.

**Comment by OptimusCrime** [comment]
I can take a look at this if @ArekX does not have the time. Mentioning myself so I'll remember. @OptimusCrime.

**Comment by alroniks** [comment]
Issue fixed, please, have a look at #14002
